src/state/dependency-tree-builder.ts

```diff
@@ -0,0 +1,42 @@
+import Service from "../model/Service.js";
+import DependencyTree, { DependencyNode } from "../model/DependencyNode.js";
+
+/**
+ * Will resolve a Service's full list of dependencies including any transitive
+ * dependencies and create a tree structure
+ */
+export class DependencyTreeBuilder {
```

Comment:
Again please add some tests for this class - performs a core part to the dependency resolution for the commands and so would pose risk if this was modified in the future of regressions being introduced

Reply:
I see this is transiently tested as part of the Inventory tests so not a biggy, might help for a documentation perspective to have a test or two for it.